### PR TITLE
docs: point readme  to new generator docs

### LINKED ISSUE #892: cleanup readme and point to new docs location
Now generator docs, on each push to master, will be published to https://www.asyncapi.com/docs/tools/generator

IMHO we should now:
- remove all content from `docs/readme.md` as it do not make sense to maintain TOC there as docs anyway should be read in the website. We can now just put short paragraph that docs for generator are renders and should be read in https://www.asyncapi.com/docs/tools/generator
- remove docs related sections from `readme.md` and just have one section `Documentation` and there short intro and link to https://www.asyncapi.com/docs/tools/generator

Thoughts?

### PR DESCRIPTION
**Pull request by Florence-Njeri**
**Description**
After the generator docs were published to the tools bucket, the links in the Readme are broken and should be read in the [website ](https://www.asyncapi.com/docs/tools/generator)

**Related issue(s)**
 `Fixes #892`

### CODE REVIEW

**Comment by Amishakumari544** [comment]
#892

**Comment by Amishakumari544** [comment]
PR fixes #892

File: docs/README.md [hunk 1]
```diff
@@ -1,18 +1,3 @@
...
-- [Template Development](template-development.md)
-- [Configuration File](configuration-file.md)
-- [Template context](template-context.md)
-- [React render engine](react-render-engine.md)
-- [Nunjucks render engine](nunjucks-render-engine.md)
-- [Hooks](hooks.md)
-- [File templates](file-templates.md)
-- [TypeScript support](typescript-support.md)
-- [Special file names](special-file-names.md)
+Read the docs on asyncapi.com. You can still read the docs from here, but some of the links may be broken.
```
Comment (pratik2315) [inline docs/README.md:3]: ## Documentation

Read the docs on https://www.asyncapi.com/docs/tools/generator. You can still read the docs from here, but some of the links may be broken.

Comment (pratik2315) [reply docs/README.md:3]: I think we should remove `Read the generator docs at https://www.asyncapi.com/docs/tools/generator` as it seems repeated (question mark!)

**Comment by pratik2315** [comment]
LGTM otherwise🚀

**Comment by Florence-Njeri** [comment]
@derberg @pratik2315 please review, I've updated PR from last review

**Review by derberg — COMMENTED** [review]
regarding changes in readme

what do you think if instead of keeping all the sections we just send people to main docs?

so readme has just one section `Documentation` where we link to https://www.asyncapi.com/docs/tools/generator

all the others could be removed 🤔

**Review by pratik2315 — DISMISSED** [review]
Lgtm👍

**Review by derberg — COMMENTED** [review]
there is one problem I see

sorry for not being precise with my comments

I meant removing of docs sections, but I think `Overview` even if it duplicates stuff a bit, it should stay right? I would just remove `To create your own template, go to section that explains [How to create a template](#how-to-create-a-template).` from the section

and regarding `## List of official generator templates`, we need it to. Templates are not yet listed on asyncapi.com/tools so we need them here at least. You could add https://github.com/asyncapi/php-template and https://github.com/asyncapi/dotnet-rabbitmq-template that are not on the list 😄 Anyway, once we remove this section, more things will have to be removed -> https://github.com/asyncapi/generator/tree/master/.github/templates-list-validator and workflow that uses it


I hope that makes sense, sorry 😅

**Comment by Florence-Njeri** [comment]
@derberg I'll make adjustments. I think I misunderstood

**Comment by derberg** [comment]
> but I think Overview even if it duplicates stuff a bit, it should stay right?

@Florence-Njeri I see you removed it, you think we do not need overview and repo description is enough?

also is there a reason for ToC removal?

**Comment by Florence-Njeri** [comment]
> > but I think Overview even if it duplicates stuff a bit, it should stay right?
> 
> @Florence-Njeri I see you removed it, you think we do not need overview and repo description is enough?
> 
> also is there a reason for ToC removal?

Hey @derberg from this issue -> https://github.com/asyncapi/generator/issues/892, I was to remove the TOC from the readme since the links may be broken and point the readers to the main docs in the website.

**Comment by derberg** [comment]
> Hey @derberg from this issue -> https://github.com/asyncapi/generator/issues/892, I was to remove the TOC from the readme since the links may be broken and point the readers to the main docs in the website.

oh, ok, now it makes sense why you wanted to remove it. Luckily in case of main readme we are covered.

@Florence-Njeri @pratik2315 so this is what "security" we have in place

1. When you look at [the source of the readme](https://raw.githubusercontent.com/asyncapi/generator/master/README.md) you see stuff like:
    ```
    <!-- toc is generated with GitHub Actions do not remove toc markers -->

    <!-- toc -->
    blablabla
    <!-- tocstop -->
    ```
2. We use `markdown-toc` package that handles the regeneration of `blablabla` part
3. We have a script that uses this tool https://github.com/asyncapi/generator/blob/master/package.json#L25
4. And this script is also reused here in https://github.com/asyncapi/generator/blob/master/package.json#L26 that we use directly in our automation:
    - when release happened -> https://github.com/asyncapi/generator/blob/master/.github/workflows/if-nodejs-version-bump.yml#L39
    - or when just docs were updated -> https://github.com/asyncapi/generator/blob/master/.github/workflows/update-docs-on-docs-commits.yml#L22

so basically you can add/remove sections in readme, but you do not have to remember about updating TOC as it will be updated by `asyncapi-bot`

--- 
so just remove sections:
- [Requirements](https://github.com/asyncapi/generator#requirements)
- [Using from the command-line interface (CLI)](https://github.com/asyncapi/generator#using-from-the-command-line-interface-cli)
- [Generator version vs Template version](https://github.com/asyncapi/generator#generator-version-vs-template-version)
- [How to create a template](https://github.com/asyncapi/generator#how-to-create-a-template)

and add new `Documentation` where you mention where our docs are rendered, and maybe mention that in case of docs contribution sources are here in docs folder

File: README.md [hunk 2]
```diff
@@ -25,8 +25,6 @@
 
 Generator is a tool that you can use to generate whatever you want basing on the AsyncAPI specification file as an input.
```
Comment (derberg) [inline README.md:26]:
```suggestion
Generator is a tool that you can use to generate whatever you want basing on the AsyncAPI specification file as an input. For more information [read the docs](https://www.asyncapi.com/docs/tools/generator).
```

**Review by derberg — COMMENTED** [review]
left one tiny suggestion as there is no link to docs in readme

also these new template, please add them to https://github.com/asyncapi/generator/blob/master/docs/template.md#generator-templates-list as for now we duplicate that info

**Review by pratik2315 — APPROVED** [review]

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
@Florence-Njeri @pratik2315 you can see what I meant that ToC in readme is updated automatically by bot -> https://github.com/asyncapi/generator/pull/932

**Comment by Florence-Njeri** [comment]
Yes, makes sense.
